### DISCUSSION TITLE
Select2_from_ajax automatically clears the selection when is not Validate Form.

### DISCUSSION THREAD

**Discussion by ngthanhluan1505** [Laravel-Backpack/community-forum#760] [Q&A (Help)]
When I submit a Form has Select2_from_ajax. It auto clears my selection if Other form fields is not valid. How to fix it?
`
CRUD::addfield([
            'name'        => 'lead_id',
            'label'       => "Lead User",
            'type'        => 'select2_from_ajax',
            'wrapper'   => [
                'class'      => 'form-group col-md-6'
                ],
            'placeholder' => '選択してください',
            'entity' => 'leadUser',
            
            'model' => 'App\Models\Project',
            'data_source'          => route('lead.index'),
            'dependencies'         => ['project_id'],
            'minimum_input_length' => 0,
            'include_all_form_fields' => true,
            'attribute'            => 'name',
            'pivot'       => true,
        ]);
`

**Comment by pxpm** [comment]
Hey @ngthanhluan1505 thanks for the report 🙏 

I am guessing the problem may be with the "dependencies". 
Does the field behave the same if it has no dependencies ? 

Also, what kind of relation is `leadUser` ? 

Cheers

**Reply by ngthanhluan1505** [reply]
HI @pxpm 
`
// in Model Project

public function leadUser()
    {
        return $this->belongsTo('App\Models\Project', 'id');
    }

`
In Case with Form is valid, it is OK and store in Database.

**Reply by pxpm** [reply]
Hey @ngthanhluan1505 I was asking if you remove the `dependencies` attribute from the field, if it would still lose the value. 

I am thinking that may be the issue, not sure.

**Reply by ngthanhluan1505** [reply]
Hi @pxpm I remove 'dependencies' => ['project_id']. But still can't solve it.

**Comment by AdamovicBoban** [comment]
Hi @ngthanhluan1505 

This is my `select2_from_ajax`, I just checked, it's not happening for me. Do you perhaps have something in the JavaScript onLoad ?

        CRUD::field('city_id')
            ->label('City')
            ->type('select2_from_array')
            ->options(City::all()->pluck('name', 'id'))
            ->allows_null(false)
            ->wrapperAttributes(['class' => 'form-group col-md-12'])
            ->tab('Order');

        CRUD::field('zip_id')
            ->label('Zip')
            ->type('select2_from_ajax')
            ->entity('zip')
            ->attribute('code')
            ->model("\App\Models\Zip")
            ->wrapper(['class' => 'form-group col-md-12'])
            ->data_source(route('api.zip'))
            ->placeholder('Zip select')
            ->minimum_input_length(0)
            ->dependencies(['city_id'])
            ->include_all_form_fields(true)
            ->method('POST')
            ->tab('Order');

        CRUD::field('street_id')
            ->type('select2_from_ajax')
            ->label('Street')
            ->entity('street')
            ->attribute('name')
            ->model("\App\Models\Street")
            ->wrapper(['class' => 'form-group col-md-6'])
            ->data_source(route('api.street'))
            ->placeholder('Street select')
            ->minimum_input_length(0)
            ->dependencies(['zip_id'])
            ->include_all_form_fields(true)
            ->method('POST')
            ->tab('Order');`

**Comment by jcastroa87** [comment]
We haven’t heard back from you in a while, so we’re closing this issue for now. If you still need assistance, please don’t hesitate to reopen the issue or create a new one.

Thank you, and take care!

Cheers.
